From e30e96a427df7b50217e8bd252ea516fc698aa73 Mon Sep 17 00:00:00 2001
From: Jan Sondhauss <jan.sondhauss@wago.com>
Date: Mon, 18 Jun 2018 13:03:46 +0200
Subject: [PATCH 181/261] net: swcfg: don't flush static entries

Flush only non-static ATU entries.
For flushing of non-static ATU entries, ATU entry state
has't to be 0.

Signed-off-by: Jan Sondhauss <jan.sondhauss@wago.com>
Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 drivers/net/phy/mv88e6321-cfg.c | 23 ++---------------------
 1 file changed, 2 insertions(+), 21 deletions(-)

diff --git a/drivers/net/phy/mv88e6321-cfg.c b/drivers/net/phy/mv88e6321-cfg.c
index 6fc3bee..40865d7 100644
--- a/drivers/net/phy/mv88e6321-cfg.c
+++ b/drivers/net/phy/mv88e6321-cfg.c
@@ -919,26 +919,7 @@ static int mv88e6321_atu_flush(struct mii_bus *bus)
 	int ret = 0;
 	u16 val = 0;
 
-	/* Set ATU-Entry State to 0 for ATU flush */
-	ret = mdiobus_read(bus, MV88E6321_GLOBAL_REG, GLOBAL_ATU_DATA);
-	if (ret < 0) {
-		pr_err("Read ATU data register failed: %d\n", ret);
-		goto out;
-	}
-
-	val = (u16) ret;
-
-	val &= ~GLOBAL_ATU_DATA_STATE_MASK;
-	val |= GLOBAL_ATU_DATA_STATE_UNUSED;
-
-	ret = mdiobus_write(bus, MV88E6321_GLOBAL_REG, GLOBAL_ATU_DATA,
-			    val);
-	if (ret) {
-		pr_err("Write ATU data failed: %d\n", ret);
-		goto out;
-	}
-
-	/* flush all ATU entries */
+	/* flush non static ATU entries */
 
 	ret = mv88e6321_wait_atu_ready(bus);
 	if (ret)
@@ -953,7 +934,7 @@ static int mv88e6321_atu_flush(struct mii_bus *bus)
 	val = (u16)ret;
 
 	val &= ~GLOBAL_ATU_OP_MASK;
-	val |= GLOBAL_ATU_OP_FLUSH_MOVE_ALL;
+	val |= GLOBAL_ATU_OP_FLUSH_MOVE_NON_STATIC;
 
 	ret = mdiobus_write(bus, MV88E6321_GLOBAL_REG, GLOBAL_ATU_OP,
 			    val);
-- 
2.7.4

